common/persistence/dataInterfaces.go

```diff
@@ -537,6 +537,23 @@ type (
 		LastEventID int64
 	}
 
+	// VersionHistoryItem contains the event id and the associated version
+	VersionHistoryItem struct {
+		EventID int64
+		Version int64
+	}
+
+	// VersionHistory provides operations on version history
+	VersionHistory struct {
+		BranchToken []byte
+		History     []VersionHistoryItem
+	}
+
+	// VersionHistories contains a set of VersionHistory
+	VersionHistories struct {
```

Comment:
we also need to store the index of current branch / version history

Reply:
Done